### PR TITLE
Fix Scheduler current time indication when groupByDate=true (T895101)

### PR DESCRIPTION
**Pull request by Yarovaya**

### CODE REVIEW

File: js/ui/scheduler/shaders/ui.scheduler.current_time_shader.horizontal.js [hunk 1]
```diff
@@ -1,35 +1,53 @@
...
-
-        if(groupCount > 1) {
-            for(let i = 1; i < groupCount; i++) {
-                const $shader = this._createShader();
-                this._customizeShader($shader, 1);
-                this._shader.push($shader);
-            }
+        for(let i = 0; i < groupCount; i++) {
+            const $shader = i === 0 ? this._$shader : this.createShader();
+            this._workSpace.isGroupedByDate() ? this._customizeGroupedByDateShader($shader, i) : this._customizeShader($shader, i);
```
Comment (AntonSermyazhko) [inline js/ui/scheduler/shaders/ui.scheduler.current_time_shader.horizontal.js]: It is better using simple if / else in this case

File: js/ui/scheduler/shaders/ui.scheduler.current_time_shader.horizontal.js [hunk 2]
```diff
@@ -1,35 +1,53 @@
...
+        const fractionPart = cellCount - integerPart;
+        const isFirstShaderPart = groupIndex === 0;
+
+        const shaderWidth = isFirstShaderPart ? this._workSpace.getIndicationWidth() : fractionPart * this._workSpace.getCellWidth();
+
+        this._applyShaderWidth($shader, shaderWidth);
+        this.applyShaderMargin($shader);
+
+        const shaderLeft = isFirstShaderPart ? this._workSpace._getCellCount() * this._workSpace.getCellWidth() * groupIndex :
+            this._workSpace.getCellWidth() * integerPart * this._workSpace._getGroupCount() + groupIndex * this._workSpace.getCellWidth();
```
Comment (AntonSermyazhko) [inline js/ui/scheduler/shaders/ui.scheduler.current_time_shader.horizontal.js]: Too complex operation for the trenary operator.
`this._workSpace` should be cached to the local var.

File: js/ui/scheduler/shaders/ui.scheduler.current_time_shader.js [hunk 3]
```diff
@@ -1,37 +1,44 @@
...
+    constructor(workSpace) {
+        this._workSpace = workSpace;
+    }
+
+    render() {
+        this._$container = this._workSpace._dateTableScrollable.$content();
 
-        this._$shader = this._createShader();
+        this._$shader = this.createShader();
         this._shader = [];
```
Comment (AntonSermyazhko) [inline js/ui/scheduler/shaders/ui.scheduler.current_time_shader.js:13]: Just a suggestion.. To my mind it is better to rename `_shader` to `_shaders`, `_shaderStore`, `_shaderCollection` etc.

File: js/ui/scheduler/shaders/ui.scheduler.current_time_shader.vertical.js [hunk 4]
```diff
@@ -1,98 +1,107 @@
...
         if(shaderHeight >= 0) {
+            if(shaderHeight > maxHeight) {
+                shaderHeight = maxHeight;
+            }
+
             this._$shader.height(shaderHeight);
+            const groupCount = this._workSpace._getGroupCount() || 1;
 
-            const groupCount = this._workspace._getGroupCount() || 1;
+            this._workSpace.isGroupedByDate() ? this._renderGroupedByDateShaderParts(groupCount, shaderHeight, maxHeight, isSolidShader) :
```
Comment (AntonSermyazhko) [inline js/ui/scheduler/shaders/ui.scheduler.current_time_shader.vertical.js]: It is better using simple if / else in this case

File: js/ui/scheduler/shaders/ui.scheduler.current_time_shader.vertical.js [hunk 5]
```diff
@@ -1,98 +1,110 @@
...
+        return this._workSpace.getGroupedStrategy().getShaderWidth(i);
+    }
 
-    clean: function() {
-        this.callBase();
+    clean() {
+        super.clean();
 
-        this._workspace && this._workspace._$allDayPanel && this._workspace._$allDayPanel.find('.' + DATE_TIME_SHADER_ALL_DAY_CLASS).remove();
+        this._workSpace && this._workSpace._$allDayPanel && this._workSpace._$allDayPanel.find('.' + DATE_TIME_SHADER_ALL_DAY_CLASS).remove();
```
Comment (AryamnovEugeniy) [inline js/ui/scheduler/shaders/ui.scheduler.current_time_shader.vertical.js:105]: `this._workSpace?._$allDayPanel?.find('.' + DATE_TIME_SHADER_ALL_DAY_CLASS).remove();` would be much shorter

**Review by AryamnovEugeniy — DISMISSED** [review]

File: js/ui/scheduler/shaders/ui.scheduler.current_time_shader.horizontal.js [hunk 6]
```diff
@@ -1,35 +1,64 @@
...
         }
-    },
+    }
+
+    _customizeShader($shader, groupIndex) {
+        const shaderWidth = this._workSpace.getIndicationWidth();
+
+        this._applyShaderWidth($shader, shaderWidth);
+
+        $shader.css('left', this._workSpace._getCellCount() * this._workSpace.getCellWidth() * (groupIndex >= 1 ? 1 : 0));
```
Comment (AntonSermyazhko) [inline js/ui/scheduler/shaders/ui.scheduler.current_time_shader.horizontal.js]: Can we simplify this:
```
if(groupIndex >= 1) {
  const workSpace = this._workSpace;
  const fullWidth = workSpace._getCellCount() * workSpace.getCellWidth();
  $shader.css('left', fullWidth);
} else {
  $shader.css('left', 0);
}
```

**Review by andrewmakarov — DISMISSED** [review]

**Review by andrewmakarov — APPROVED** [review]

File: js/ui/scheduler/shaders/ui.scheduler.current_time_shader.vertical.js (hunk 5 above)
Comment (AntonSermyazhko) [reply js/ui/scheduler/shaders/ui.scheduler.current_time_shader.vertical.js:105]: Maybe use interpolation here: `'.' + DATE_TIME_SHADER_ALL_DAY_CLASS` ?

**Review by AntonSermyazhko — APPROVED** [review]

**Pull request merged**